### PR TITLE
Fix for #38838

### LINKED ISSUE #38838: Failing to remove nginx
### Description of Issue/Question
Hi there,

I am playing with SaltStack, trying to get a feel for everything. I've ran into what I think could possibly be a bug - removing nginx does not work correctly. All involved machines are running fully updated Ubuntu 16.04 and all packages are current. For what it's worth, the salt minions are all running as VMware virtual machines with 4GB of RAM each. None of the minions have any extra apt repos besides SaltStack (no DotDeb or anything like that).

### Setup
This is my top.sls file:
```
base:
  '*':
    - core
```

This is my core/init.sls file:
```
remove-things:
  pkg.removed:
    - name: nginx
```

### Steps to Reproduce Issue
- Install nginx on minion
- On master, run sudo salt 'vm1' state.apply core
- First result:
```
vm1:
----------
          ID: remove-things
    Function: pkg.removed
        Name: nginx
      Result: False
     Comment: The following packages failed to remove: nginx.
     Started: 21:21:08.401320
    Duration: 1315.336 ms
     Changes:
              ----------
              nginx:
                  ----------
                  new:
                      1
                  old:
                      1.10.0-0ubuntu0.16.04.4

```

It does succeed at removing the nginx package, but claims to have failed.

Now, run it again:
- On master, run sudo salt 'vm1' state.apply core
```

vm1:
----------
          ID: remove-things
    Function: pkg.removed
        Name: nginx
      Result: False
     Comment: The following packages failed to remove: nginx.
     Started: 21:22:14.792268
    Duration: 977.843 ms
     Changes:

```

The exact same process does NOT fail with simple packages like "sl" and "cowsay".

### Versions Report
```
root@nebula:/srv/salt# salt --versions-report
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-59-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

### PR DESCRIPTION
**Pull request by samodid**
### What does this PR do?
fix behavior of aptpkg.list_pkgs function. Now it works as in description "List the packages currently INSTALLED in a dict" 
### What issues does this PR fix or reference?
fix #38838
### Previous Behavior
return wrong data
check with #38838 issue

### New Behavior

returns info only about installed packages

### Tests written?

No

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/aptpkg.py [hunk 1]
```diff
@@ -1401,7 +1401,9 @@ def list_pkgs(versions_as_list=False,
         for realpkg, provides in six.iteritems(virtpkgs_all):
             # grep-available returns info on all virtual packages. Ignore any
             # virtual packages that do not have the real package installed.
-            if realpkg in ret['installed']:
+            # _get_virtual() do not use grep-available if HAS_APT is true
+            # so we can skip loop below
+            if not HAS_APT and realpkg in ret['installed']:
```
Comment (DmitryKuzmenko) [inline salt/modules/aptpkg.py]: Now if `not HAS_APT` all the `if not removed` case does nothing.
I mean it's better to add the `not HAS_APT` case to the enclosing `if not removed` 11 lines above:
```python
if not removed and not HAS_APT:
```

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

File: salt/modules/aptpkg.py (hunk 1 above)
Comment (samodid) [reply salt/modules/aptpkg.py]: Hello Dmitry,
thanks for your comment and suggestion. It makes sense. 

I've checked that change - it works.  

```
# salt-call pkg.list_pkgs --output=json -l quiet |  jq '."local" |=with_entries(select( .key | test("nginx") ))'
{
  "local": {
    "nginx-core": "1.10.3-0ubuntu0.16.04.2",
    "nginx-common": "1.10.3-0ubuntu0.16.04.2"
  }
}
```

**Comment by cachedout** [comment]
We'll need @DmitryKuzmenko to re-review this.

**Review by DmitryKuzmenko — COMMENTED** [review]
Looks good to me now.
But I'm not completely understand all the apt/dpkg dancing happening in that module. =(

**Review by DmitryKuzmenko — APPROVED** [review]

**Pull request merged**